HACKING.md

```diff
@@ -222,10 +222,55 @@ maybe you need to replace system installed snap-seccomp with the one aligned to
 you are testing. To do this, simply backup /usr/lib/snapd/snap-seccomp and overwrite it with 
 the testing one. Don't forget to rollback to the original when finish testing)
 
+### Running nested tests
+
+Nested tests are used to validate features which cannot be tested on regular tests.
+
+The nested test suites work different from the other test suites in snapd. In this case each test runs in a new image
+which is created following the rules defined for the test.
+
+The nested tests are executed using spread tool. See the following examples using the qemu and google backends.
+
+    . qemu: spread qemu-nested:ubuntu-20.04-64:tests/nested/core20/tpm
+    . google: spread google-nested:ubuntu-20.04-64:tests/nested/core20/tpm
+
+The nested system in all the cases is selected based on the host system. The folloing lines show the relation between host and nested systemd (same applies for classic nested tests):
+
+    . ubuntu-16.04-64 => ubuntu-core-16-64
+    . ubuntu-18.04-64 => ubuntu-core-18-64
+    . ubuntu-20.04-64 => ubuntu-core-20-64
+
+The tools used for creating and hosting the nested vms are:
+
+    . ubuntu-image snap is used to building the images
+    . QEMU is used for the virtualization (with kvm acceleration)
+
+Nested test suite is composed by the following 4 suites:
+
+    classic: the nested suite contains an image of a classic system downloaded from cloud-images.ubuntu.com 
+    core: it tests a core nested system and the images are generated by using ubuntu-image snap
+    core20: this is similar to core suite but tests on it are focused on UC20
+    manual: tests on this suite create a non generic image with spedific conditions
+
+The nested suites use some environment variables to configure the suite and the tests inside it, the most important ones are the described bellow:
+
+    SPREAD_WORK_DIR: It is path to the directory where all the nested assets and images are stored
```

Comment:
We should not use a variable that may plausibly used by spread itself. Those are our own variables and should be called accordingly.

Reply:
updated